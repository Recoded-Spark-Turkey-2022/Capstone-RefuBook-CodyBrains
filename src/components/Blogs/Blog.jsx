import React from 'react';

const Blog = ({item,isSmallCard}) => {
    return (
        <div
            className="max-w-sm bg-white shadow-md
             dark:bg-gray-800 mx-3 mt-4">
            <a href="!#">
                <img className="" src={item.cover} alt={item.title}/>
            </a>
            <div className="p-5">
                <a href="!#">
                    <h6 className="mb-2 text-2xl font-bold tracking-tight text-gray-900 dark:text-white"
                    >
                         {isSmallCard ? item.title.substring(0, 15) : item.title.substring(0, 20)}
                    </h6>
                </a>
                <p className="mb-3 font-normal text-gray-700 dark:text-gray-400">
                    {item.text.substring(0, 50)}
                </p>


                <div className="flex items-center space-x-4">
                    <img className="w-10 h-10 rounded-full" src={item.avatar} alt={item.title}/>
                    <div className="text-refubookBlue">
                        <div>{item.userName}</div>
                    </div>
                </div>

            </div>
        </div>
    );
};

export default Blog;
