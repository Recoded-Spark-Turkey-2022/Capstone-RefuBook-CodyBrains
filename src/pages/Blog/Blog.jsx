
import { Link } from 'react-router-dom';
import { Pagination } from 'swiper';
import { Swiper, SwiperSlide } from 'swiper/react';
import blogsData from '../../data/blogsData';

const Blog = () => {
  const firstFourBlogs = blogsData.slice(0, 4);

  return (

    <div className="bg-refubookWhite">
      <div className="  py-16 px-8 mx-auto container">
        <h2 className="sr-only">Blog Posts</h2>
        <Swiper
          spaceBetween={15}
          pagination={{
            clickable: true,
            dynamicBullets: true,
          }}
          modules={[Pagination]}
          className="w-full "
          loop
          breakpoints={{
            320: {
              slidesPerView: 1,
              spaceBetween: 10,
            },
            640: {
              slidesPerView: 1,
              spaceBetween: 5,
            },
            768: {
              slidesPerView: 2,
              spaceBetween: 5,
            },
            1024: {
              slidesPerView: 3,
              spaceBetween: 5,
            },
            1280: {
              slidesPerView: 3,
              spaceBetween: 5,
            },
          }}
        >
          {blogsData.map((item) => {
            return (
              <SwiperSlide
                key={item.id}
                className=" rounded-lg border border-gray-100 shadow-sm"
              >
                <img
                  src={item.cover}
                  alt={item.title}
                  className="h-56 w-full object-cover"
                />

                <div className="p-6">
                  <Link to={`blogs/${item.id}`}>
                    <h3 className="text-md font-medium ">{item.title}</h3>
                  </Link>
                  <p className="mt-2 text-sm leading-relaxed  truncate">
                    {item.text}
                  </p>
                  <div className="group mt-4 inline-flex items-center gap-1 text-sm font-medium">
                    <img
                      src={item.avatar}
                      alt=""
                      className="w-10 h-10 rounded-full"
                    />
                    <p className="block transition group-hover:translate-x-0.5 ">
                      {item.userName}
                    </p>
                  </div>
                </div>
              </SwiperSlide>
            );
          })}
          ;
        </Swiper>
      </div>
      <div className="grid grid-cols-2  md:grid-cols-4 gap-4 md:gap-6 p-4">
        {firstFourBlogs.map((item) => {
          return (
            <div
              className="relative group shadow-md bg-refubookWhite border border-refubookGray rounded-lg flex flex-col overflow-hidden "
              key={item.id}
            >
              <img src={item.cover} alt={item.title} />

              <div className=" p-4 space-y-2 flex flex-col ">
                <p className="text-xs md:text-md font-medium text-gray-900">
                  <Link to={`blogs/${item.id}`}>
                    <span aria-hidden="true" className="absolute inset-0" />
                    {item.title}
                  </Link>
                </p>
                <p className="text-sm text-gray-500 truncate">{item.text}</p>
                <div className="flex-1 flex flex-col justify-center">
                  <img
                    src={item.avatar}
                    alt=""
                    className="w-10 h-10 rounded-full"
                  />
                  <p className="text-sm text-gray-500">{item.userName}</p>
                </div>
              </div>
            </div>
          );
        })}
      </div>
    </div>
  );
};

export default Blog;

