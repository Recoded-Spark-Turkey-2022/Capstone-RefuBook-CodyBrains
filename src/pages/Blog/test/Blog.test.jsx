import renderer from 'react-test-renderer';
import Blog from '../Blog';

jest.mock('react-router-dom', () => ({
  Link: 'Link',
}));

jest.mock('firebase/firestore', () => ({
  ...jest.requireActual('firebase/firestore'),
  collection: () => jest.fn(),
  getDocs: () => jest.fn(),
}));

jest.mock('html-react-parser', () => jest.fn());

it('renders correctly', () => {
  const tree = renderer.create(<Blog />).toJSON();
  expect(tree).toMatchSnapshot();
});
