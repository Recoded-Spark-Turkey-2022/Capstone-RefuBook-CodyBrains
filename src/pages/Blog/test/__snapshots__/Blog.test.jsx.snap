// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="bg-refubookWhite"
>
  <div
    className="  py-16 px-8 mx-auto container"
  >
    <h2
      className="sr-only"
    >
      Blog Posts
    </h2>
    <div
      className="swiper w-full"
    >
      <div
        className="swiper-wrapper"
      >
        <div
          className="swiper-slide rounded-lg border border-gray-100 shadow-sm swiper-slide-duplicate"
          data-swiper-slide-index={3}
        >
          <img
            alt="The best way to learn a language from locals as a refugee"
            className="h-56 w-full object-cover"
            src="blog-cover.jpg"
          />
          <div
            className="p-6"
          >
            <Link
              to="blogs/4"
            >
              <h3
                className="text-md font-medium "
              >
                The best way to learn a language from locals as a refugee
              </h3>
            </Link>
            <p
              className="mt-2 text-sm leading-relaxed  truncate"
            >
              There are several ways to learn a language from locals as a refugee: Enroll in a language class: Many communities offer language classes for refugees and immigrants. These classes are usually taught by trained instructors and can be a good way to learn the basics of a new language. Join a language exchange program: Language exchange programs match you with a native speaker of the language you want to learn. You can meet with them in person or communicate online and practice speaking and listening to the language. Find a tutor: You can also find a tutor who can help you learn the language one-on-one. This can be a more personalized approach to learning and may be more effective for some people. Immerse yourself in the language: Surround yourself with native speakers and try to communicate in the language as much as possible. This can be challenging, but it can be an effective way to learn a language quickly.Use language learning resources: There are many resources available online and in print that can help you learn a language, including language learning apps, podcasts, and textbooks. These can be a helpful supplement to other language learning methods. Its important to be patient and consistent when learning a new language. It takes time and practice to become proficient, so dont get discouraged if you dont see progress right away. Keep practicing and you will improve over time.
            </p>
            <div
              className="group mt-4 inline-flex items-center gap-1 text-sm font-medium"
            >
              <img
                alt=""
                className="w-10 h-10 rounded-full"
                src="avatar.png"
              />
              <p
                className="block transition group-hover:translate-x-0.5 "
              >
                ahmad
              </p>
            </div>
          </div>
        </div>
        <div
          className="swiper-slide rounded-lg border border-gray-100 shadow-sm"
          data-swiper-slide-index={0}
        >
          <img
            alt="Difficulties faced by immigrants"
            className="h-56 w-full object-cover"
            src="blog-cover.jpg"
          />
          <div
            className="p-6"
          >
            <Link
              to="blogs/1"
            >
              <h3
                className="text-md font-medium "
              >
                Difficulties faced by immigrants
              </h3>
            </Link>
            <p
              className="mt-2 text-sm leading-relaxed  truncate"
            >
              Leaving ones home country and immigrating to a new place can be a challenging and emotional experience, especially for refugees who may be fleeing conflict, persecution, or other difficult circumstances
            </p>
            <div
              className="group mt-4 inline-flex items-center gap-1 text-sm font-medium"
            >
              <img
                alt=""
                className="w-10 h-10 rounded-full"
                src="avatar.png"
              />
              <p
                className="block transition group-hover:translate-x-0.5 "
              >
                Tahsin
              </p>
            </div>
          </div>
        </div>
        <div
          className="swiper-slide rounded-lg border border-gray-100 shadow-sm"
          data-swiper-slide-index={1}
        >
          <img
            alt="Nours immigration story"
            className="h-56 w-full object-cover"
            src="blog-cover.jpg"
          />
          <div
            className="p-6"
          >
            <Link
              to="blogs/2"
            >
              <h3
                className="text-md font-medium "
              >
                Nours immigration story
              </h3>
            </Link>
            <p
              className="mt-2 text-sm leading-relaxed  truncate"
            >
              Nour was a young Syrian girl who lived with her parents and two younger brothers in a small village near the border of Turkey. One day, their peaceful village was attacked by a group of armed men, and Nours family was forced to flee for their lives.They traveled for many days, walking through dangerous territories and enduring harsh conditions, until they finally reached the border of Syria and Turkey. When they arrived, they were exhausted and hungry, but they were also relieved to have escaped the violence of their homeland.As they entered Turkey, they were welcomed by a group of kind-hearted volunteers who offered them food, water, and shelter. Nour and her family were grateful for the generosity and warmth of their new friends, and they knew that they had finally found a safe place to call home.Over the next few years, Nour and her family settled into their new life in Turkey, and although they missed their home and their loved ones, they were grateful for the opportunity to start anew. And although their journey had been difficult and filled with challenges, they knew that they were stronger because of it.
            </p>
            <div
              className="group mt-4 inline-flex items-center gap-1 text-sm font-medium"
            >
              <img
                alt=""
                className="w-10 h-10 rounded-full"
                src="avatar.png"
              />
              <p
                className="block transition group-hover:translate-x-0.5 "
              >
                nour
              </p>
            </div>
          </div>
        </div>
        <div
          className="swiper-slide rounded-lg border border-gray-100 shadow-sm"
          data-swiper-slide-index={2}
        >
          <img
            alt="Directions to do as a refugee moving to a new country"
            className="h-56 w-full object-cover"
            src="blog-cover.jpg"
          />
          <div
            className="p-6"
          >
            <Link
              to="blogs/3"
            >
              <h3
                className="text-md font-medium "
              >
                Directions to do as a refugee moving to a new country
              </h3>
            </Link>
            <p
              className="mt-2 text-sm leading-relaxed  truncate"
            >
              If you are a refugee moving to a new country, there are a few steps you can take to help you adjust to your new environment and get settled in your new home: Learn about the country you are moving to: Research the culture, customs, and laws of the country you are moving to. This will help you understand the expectations and norms of your new home. Find a place to live: Look for housing options that fit your budget and needs. Consider contacting a local refugee agency or community organization for assistance with finding housing. Enroll your children in school: If you have children, make sure to enroll them in school as soon as possible. This will help them adjust to their new environment and continue their education. Learn the language: If the country you are moving to speaks a different language, try to learn as much as you can before you arrive. This will make it easier for you to communicate with others and access services. Find medical care: Make sure to find a healthcare provider and get any necessary medical treatment. This may involve finding a new primary care doctor and getting any necessary vaccinations. Connect with the community: Join local groups or organizations that can help you connect with other refugees and members of the community. This can help you feel more connected and supported in your new home. Seek out resources and support: There may be organizations or agencies in your new community that offer assistance to refugees, such as legal assistance, language classes, and job training. Look for these resources and take advantage of them to help you succeed in your new home.
            </p>
            <div
              className="group mt-4 inline-flex items-center gap-1 text-sm font-medium"
            >
              <img
                alt=""
                className="w-10 h-10 rounded-full"
                src="avatar.png"
              />
              <p
                className="block transition group-hover:translate-x-0.5 "
              >
                alparslan
              </p>
            </div>
          </div>
        </div>
        <div
          className="swiper-slide rounded-lg border border-gray-100 shadow-sm"
          data-swiper-slide-index={3}
        >
          <img
            alt="The best way to learn a language from locals as a refugee"
            className="h-56 w-full object-cover"
            src="blog-cover.jpg"
          />
          <div
            className="p-6"
          >
            <Link
              to="blogs/4"
            >
              <h3
                className="text-md font-medium "
              >
                The best way to learn a language from locals as a refugee
              </h3>
            </Link>
            <p
              className="mt-2 text-sm leading-relaxed  truncate"
            >
              There are several ways to learn a language from locals as a refugee: Enroll in a language class: Many communities offer language classes for refugees and immigrants. These classes are usually taught by trained instructors and can be a good way to learn the basics of a new language. Join a language exchange program: Language exchange programs match you with a native speaker of the language you want to learn. You can meet with them in person or communicate online and practice speaking and listening to the language. Find a tutor: You can also find a tutor who can help you learn the language one-on-one. This can be a more personalized approach to learning and may be more effective for some people. Immerse yourself in the language: Surround yourself with native speakers and try to communicate in the language as much as possible. This can be challenging, but it can be an effective way to learn a language quickly.Use language learning resources: There are many resources available online and in print that can help you learn a language, including language learning apps, podcasts, and textbooks. These can be a helpful supplement to other language learning methods. Its important to be patient and consistent when learning a new language. It takes time and practice to become proficient, so dont get discouraged if you dont see progress right away. Keep practicing and you will improve over time.
            </p>
            <div
              className="group mt-4 inline-flex items-center gap-1 text-sm font-medium"
            >
              <img
                alt=""
                className="w-10 h-10 rounded-full"
                src="avatar.png"
              />
              <p
                className="block transition group-hover:translate-x-0.5 "
              >
                ahmad
              </p>
            </div>
          </div>
        </div>
        <div
          className="swiper-slide rounded-lg border border-gray-100 shadow-sm swiper-slide-duplicate"
          data-swiper-slide-index={0}
        >
          <img
            alt="Difficulties faced by immigrants"
            className="h-56 w-full object-cover"
            src="blog-cover.jpg"
          />
          <div
            className="p-6"
          >
            <Link
              to="blogs/1"
            >
              <h3
                className="text-md font-medium "
              >
                Difficulties faced by immigrants
              </h3>
            </Link>
            <p
              className="mt-2 text-sm leading-relaxed  truncate"
            >
              Leaving ones home country and immigrating to a new place can be a challenging and emotional experience, especially for refugees who may be fleeing conflict, persecution, or other difficult circumstances
            </p>
            <div
              className="group mt-4 inline-flex items-center gap-1 text-sm font-medium"
            >
              <img
                alt=""
                className="w-10 h-10 rounded-full"
                src="avatar.png"
              />
              <p
                className="block transition group-hover:translate-x-0.5 "
              >
                Tahsin
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        className="swiper-pagination"
      />
      ;
    </div>
  </div>
  <div
    className="grid grid-cols-2  md:grid-cols-4 gap-4 md:gap-6 p-4"
  >
    <div
      className="relative group shadow-md bg-refubookWhite border border-refubookGray rounded-lg flex flex-col overflow-hidden "
    >
      <img
        alt="Difficulties faced by immigrants"
        src="blog-cover.jpg"
      />
      <div
        className=" p-4 space-y-2 flex flex-col "
      >
        <p
          className="text-xs md:text-md font-medium text-gray-900"
        >
          <Link
            to="blogs/1"
          >
            <span
              aria-hidden="true"
              className="absolute inset-0"
            />
            Difficulties faced by immigrants
          </Link>
        </p>
        <p
          className="text-sm text-gray-500 truncate"
        >
          Leaving ones home country and immigrating to a new place can be a challenging and emotional experience, especially for refugees who may be fleeing conflict, persecution, or other difficult circumstances
        </p>
        <div
          className="flex-1 flex flex-col justify-center"
        >
          <img
            alt=""
            className="w-10 h-10 rounded-full"
            src="avatar.png"
          />
          <p
            className="text-sm text-gray-500"
          >
            Tahsin
          </p>
        </div>
      </div>
    </div>
    <div
      className="relative group shadow-md bg-refubookWhite border border-refubookGray rounded-lg flex flex-col overflow-hidden "
    >
      <img
        alt="Nours immigration story"
        src="blog-cover.jpg"
      />
      <div
        className=" p-4 space-y-2 flex flex-col "
      >
        <p
          className="text-xs md:text-md font-medium text-gray-900"
        >
          <Link
            to="blogs/2"
          >
            <span
              aria-hidden="true"
              className="absolute inset-0"
            />
            Nours immigration story
          </Link>
        </p>
        <p
          className="text-sm text-gray-500 truncate"
        >
          Nour was a young Syrian girl who lived with her parents and two younger brothers in a small village near the border of Turkey. One day, their peaceful village was attacked by a group of armed men, and Nours family was forced to flee for their lives.They traveled for many days, walking through dangerous territories and enduring harsh conditions, until they finally reached the border of Syria and Turkey. When they arrived, they were exhausted and hungry, but they were also relieved to have escaped the violence of their homeland.As they entered Turkey, they were welcomed by a group of kind-hearted volunteers who offered them food, water, and shelter. Nour and her family were grateful for the generosity and warmth of their new friends, and they knew that they had finally found a safe place to call home.Over the next few years, Nour and her family settled into their new life in Turkey, and although they missed their home and their loved ones, they were grateful for the opportunity to start anew. And although their journey had been difficult and filled with challenges, they knew that they were stronger because of it.
        </p>
        <div
          className="flex-1 flex flex-col justify-center"
        >
          <img
            alt=""
            className="w-10 h-10 rounded-full"
            src="avatar.png"
          />
          <p
            className="text-sm text-gray-500"
          >
            nour
          </p>
        </div>
      </div>
    </div>
    <div
      className="relative group shadow-md bg-refubookWhite border border-refubookGray rounded-lg flex flex-col overflow-hidden "
    >
      <img
        alt="Directions to do as a refugee moving to a new country"
        src="blog-cover.jpg"
      />
      <div
        className=" p-4 space-y-2 flex flex-col "
      >
        <p
          className="text-xs md:text-md font-medium text-gray-900"
        >
          <Link
            to="blogs/3"
          >
            <span
              aria-hidden="true"
              className="absolute inset-0"
            />
            Directions to do as a refugee moving to a new country
          </Link>
        </p>
        <p
          className="text-sm text-gray-500 truncate"
        >
          If you are a refugee moving to a new country, there are a few steps you can take to help you adjust to your new environment and get settled in your new home: Learn about the country you are moving to: Research the culture, customs, and laws of the country you are moving to. This will help you understand the expectations and norms of your new home. Find a place to live: Look for housing options that fit your budget and needs. Consider contacting a local refugee agency or community organization for assistance with finding housing. Enroll your children in school: If you have children, make sure to enroll them in school as soon as possible. This will help them adjust to their new environment and continue their education. Learn the language: If the country you are moving to speaks a different language, try to learn as much as you can before you arrive. This will make it easier for you to communicate with others and access services. Find medical care: Make sure to find a healthcare provider and get any necessary medical treatment. This may involve finding a new primary care doctor and getting any necessary vaccinations. Connect with the community: Join local groups or organizations that can help you connect with other refugees and members of the community. This can help you feel more connected and supported in your new home. Seek out resources and support: There may be organizations or agencies in your new community that offer assistance to refugees, such as legal assistance, language classes, and job training. Look for these resources and take advantage of them to help you succeed in your new home.
        </p>
        <div
          className="flex-1 flex flex-col justify-center"
        >
          <img
            alt=""
            className="w-10 h-10 rounded-full"
            src="avatar.png"
          />
          <p
            className="text-sm text-gray-500"
          >
            alparslan
          </p>
        </div>
      </div>
    </div>
    <div
      className="relative group shadow-md bg-refubookWhite border border-refubookGray rounded-lg flex flex-col overflow-hidden "
    >
      <img
        alt="The best way to learn a language from locals as a refugee"
        src="blog-cover.jpg"
      />
      <div
        className=" p-4 space-y-2 flex flex-col "
      >
        <p
          className="text-xs md:text-md font-medium text-gray-900"
        >
          <Link
            to="blogs/4"
          >
            <span
              aria-hidden="true"
              className="absolute inset-0"
            />
            The best way to learn a language from locals as a refugee
          </Link>
        </p>
        <p
          className="text-sm text-gray-500 truncate"
        >
          There are several ways to learn a language from locals as a refugee: Enroll in a language class: Many communities offer language classes for refugees and immigrants. These classes are usually taught by trained instructors and can be a good way to learn the basics of a new language. Join a language exchange program: Language exchange programs match you with a native speaker of the language you want to learn. You can meet with them in person or communicate online and practice speaking and listening to the language. Find a tutor: You can also find a tutor who can help you learn the language one-on-one. This can be a more personalized approach to learning and may be more effective for some people. Immerse yourself in the language: Surround yourself with native speakers and try to communicate in the language as much as possible. This can be challenging, but it can be an effective way to learn a language quickly.Use language learning resources: There are many resources available online and in print that can help you learn a language, including language learning apps, podcasts, and textbooks. These can be a helpful supplement to other language learning methods. Its important to be patient and consistent when learning a new language. It takes time and practice to become proficient, so dont get discouraged if you dont see progress right away. Keep practicing and you will improve over time.
        </p>
        <div
          className="flex-1 flex flex-col justify-center"
        >
          <img
            alt=""
            className="w-10 h-10 rounded-full"
            src="avatar.png"
          />
          <p
            className="text-sm text-gray-500"
          >
            ahmad
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
